pkg/client/client.go

```diff
@@ -974,7 +974,21 @@ func (c *Client) WaitForDaemonSetRollout(ds *appsv1.DaemonSet) error {
 				d.Status.DesiredNumberScheduled, d.Status.UpdatedNumberScheduled)
 			return false, nil
 		}
-		if d.Status.NumberUnavailable != 0 {
+
+		var nodeNotReadyCount int32
+		nodeList, err := c.kclient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		nodes := nodeList.Items
+		for _, node := range nodes {
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady {
```

Comment:
Won't this only catch nodes that are in ready state? what about nodes that are under other conditions (eg disk pressure) are we happy to ignore them?